core/src/main/java/bisq/core/btc/wallet/BtcWalletService.java

```diff
@@ -150,7 +150,7 @@ String getWalletAsString(boolean includePrivKeys) {
 
     public Transaction completePreparedBurnBsqTx(Transaction preparedBurnFeeTx, byte[] opReturnData)
             throws WalletException, InsufficientMoneyException, TransactionVerificationException {
-        return completePreparedProposalTx(preparedBurnFeeTx, opReturnData, null, null);
+        return completePreparedBsqTx(preparedBurnFeeTx, false, opReturnData);
```

Comment:
That method is also called from BaseProposalFactory for proposals. 
I need more time to check if completePreparedBurnBsqTx could be used for proposals.

Reply:
Not sure if that fix is worth the effort and risk atm, it is a very unlikely event that the input matches exactly the miner fee. And I assume it would fail anyway on the BitcoinJ side at least for the case with no outputs. Maybe we postpone that for a later time as it is not relevant for consensus anyway?

Reply:
Indeed, it's almost impossible to make the burn fee match mining fee and completely spend the remaining BSQ in an output. I had to write some code just to make that happen.

Since there is an output (the opReturn) I think the tx might even be standard. I thought that wasn't the case at first but reading up I can't find anything saying otherwise. Best not merge this PR then.

Reply:
About your comments, the difference between `completePreparedProposalTx` and `completePreparedBsqTx` is the reimbursement option so they should work the same for non reimbursement proposals except that `completePreparedBsqTx` also handles the no output case. I think we have a bit of a problem with too much copy pasted code here.

Reply:
Yes I agree, the code need to be cleaned up and refactored, I am just too scary to do that now. Needs much care and testing. I prefer to focus on issues which are consensus relevant as there we cannot change later anymore (or it becomes much harder).

Reply:
Maybe lets close it for now.